fellow-submissions/devonte-duncan.md

```diff
@@ -0,0 +1,154 @@
+# Problem Set 2.4 - Scope
+
+**1. How do variables declared with `let`, `const`, and `var` differ? Be sure to touch on _scope_, _reassignment_, and _hoisting_ for each.**
+
+   To begin with, the declaration of all three of these variable types are automatically 
+   hoisted by javascript right before runtime, simply because they are variables.
+   As for their differences, variables declared with `const`, are designed to
+   be constant and therefore, JavaScript does not allow it's value to be reassigned.
+   On the other hand, using `let` to declare a variable does allow you to reassign 
+   its value later on. `let` and `const` are scoped variables, which means that its
+   existence can only be detected within its parent scope, whether that be global 
+   or local. As for `var` its properties are the same as `let`, except for the fact
+   that its scope is global regardless of the location of its declaration.
+
+**2. What will the following code log? Why?**
+    ```javascript
+    let a = 'outer';
+
+    function testScope() {
+      let a = 'inner';
+      console.log(a);
+    }
+
+    console.log(a);
+    testScope();
+    console.log(a);
+    ```
+    This above code will log:
+        "outer"
+        "undefined"
+        "outer"
+        
+    This is logged because the string variable `a` is declared both in the global 
+    scope as well as in the local scope of the `testScope` function. The value of `a` 
+    that gets logged on lines 24 and 26 is "outer" because that's the onlt value of `a`
+    that it has access to. Meanwhile "undefined" is logged because the function `testScope`
+    is referenced, however there is no return value within it, thus "undefined" is logged.
+
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+    This above code will log:
+        "undefined"
+        "ReferenceError: a is not defined"
+    
+    This is logged because the string variable `a` is declared only in the local scope 
+    of the `hello` function, thus trying to log `a` to the console will result in an error.
+    "undefined" is logged because the function `hello` is referenced, however there is no return 
+    value within it, thus "undefined" is logged.
```

Comment:
Yes! great explanation. Only small revision. `undefined` isn't _logged_ because there is no `console.log` statement in the body of the function. However, `undefined` is _returned_.